### ISSUE TITLE
Select component prevent to build typescript application

### ISSUE THREAD

**Issue by lamilsage** [Sage/carbon#3572]
### Current behaviour
I created a new react application with typescript and carbon-react.
Adding code source for select component taken from https://carbon.sage.com/?path=/docs/design-system-select--basic
I am not able to build my application

### Expected behaviour
Using select component in type script application should not prevent to build it

### Reproducible example
What I did to create the TS react app :
```
npx create-react-app simplereactapp  --template typescript
npm install --save carbon-react carbon-factory i18n-js styled-components
npm install --save node-sass
```

Then I added the following block to app.tsx

```
import { Select, Option } from "carbon-react/lib/components/select";

...
function App() {
  return (
  ...
 <Select id="controlled" name="controlled" value="1" onChange={onChangeHandler}>
           <Option text="Amber" value="1" />
           <Option text="Black" value="2" />
           <Option text="Blue" value="3" />
           <Option text="Brown" value="4" />
           <Option text="Green" value="5" />
           <Option text="Orange" value="6" />
           <Option text="Pink" value="7" />
           <Option text="Purple" value="8" />
           <Option text="Red" value="9" />
           <Option text="White" value="10" />
           <Option text="Yellow" value="11" />
 </Select>
 ...
 );
```

Then run a npm run build

```
 npm run build

> simplereactapp@0.1.0 build C:\src\simplereactapp
> react-scripts build

Creating an optimized production build...
Failed to compile.

C:/src/simplereactapp/src/App.tsx
TypeScript error in C:/src/simplereactapp/src/App.tsx(30,9):
No overload matches this call.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.
  Overload 2 of 2, '(props: PropsWithChildren<SimpleSelectProps>, context?: any): ReactElement<any, any> | Component<SimpleSelectProps, any, any> | null', gave the following error.
    Type 'Element' is not assignable to type 'ComponentType<OptionProps>'.
      Type 'Element' is not assignable to type 'FunctionComponent<OptionProps>'.  TS2769

    28 |           Learn React
    29 |         </a>
    30 |         <Select id="controlled" name="controlled" value="1" onChange={onChangeHandler}>
       |         ^
    31 |           <Option text="Amber" value="1" />
    32 |           <Option text="Black" value="2" />
    33 |           <Option text="Blue" value="3" />


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! simplereactapp@0.1.0 build: `react-scripts build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the simplereactapp@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\lamil\AppData\Roaming\npm-cache\_logs\2021-01-11T15_42_58_125Z-debug.log
```

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 56.1.0
| carbon-factory   | 11.0.1
| react-scripts    | 4.0.1
| React            | 17.0.1
| Browser          |
| npm              | 6.14.9
| Operating System | Windows 10
| typescript       | 4.1.3

**Comment by sjohnson-sage** [comment]
FE-3601

**Comment by grabkowski** [comment]
fixed in #3593
